aten/src/ATen/native/native_functions.yaml

```diff
@@ -2348,13 +2348,23 @@
   # - Tensor Tensor::index(ArrayRef<TensorIndex> indices)
   # - Tensor Tensor::index(std::initializer_list<TensorIndex> indices)
 
+- func: index_copy.out(Tensor self, int dim, Tensor index, Tensor source, *, Tensor(a!) out) -> Tensor(a!)
+  structured: True
+  variants: function
+  precomputed:
+  - dim -> int dim
+  dispatch:
+    CPU, CUDA: index_copy_out
+
 - func: index_copy_(Tensor(a!) self, int dim, Tensor index, Tensor source) -> Tensor(a!)
   variants: method
+  structured_delegate: index_copy.out
   dispatch:
     CompositeExplicitAutograd: index_copy_
```

Comment:
Is there a reason we want to keep the `CompositeExplicitAutograd` entry around?

On a related note, I think this PR will need XLA-side changes, since right now XLA just lowers `index_copy_()` directly. cc @JackCaoG

Reply:
@miladm Can you help on this one? I think we just need to lower `index_copy` too.

Reply:
Thanks @bdhirsh. I will take a look at this and circle back here.

Reply:
> Is there a reason we want to keep the CompositeExplicitAutograd entry around?

@bdhirsh - Thanks for the question. I removed it from the `native_functions.yaml` file, and everything seems to work. I'm not aware of any reason why we might want to keep it, do you think it's OK to remove it? (I've pushed a commit removing it, but can revert if you suggest otherwise).

Reply:
It looks like before this PR, `index_copy_()` was a `CompositeExplicitAutograd` kernel that just called into `_index_copy_()`. We're already deleting the private `_index_copy_()` function, so I don't think it's any more BC-breaking to additionally delete this composite kernel.

Before this PR, a backend would get `index_copy()` and `index_copy_()` for free if they implemented the private `_index_copy_()`. And now, they get both for free if they implement `index_copy(out=...)` (or in XLA's case, they get all of them if they just implement `index_copy()`.

Reply:
@bdhirsh this [PR](https://github.com/pytorch/xla/pull/3331) is tracking the fix for `index_copy`. I am working on it today.